parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -234,18 +237,38 @@ public Encoding getEncoding(org.apache.parquet.column.Encoding encoding) {
     return Encoding.valueOf(encoding.name());
   }
 
-  public static Statistics toParquetStatistics(org.apache.parquet.column.statistics.Statistics statistics) {
+  public static Statistics toParquetStatistics(
+      String createdBy, org.apache.parquet.column.statistics.Statistics statistics) {
     Statistics stats = new Statistics();
     if (!statistics.isEmpty()) {
       stats.setNull_count(statistics.getNumNulls());
-      if(statistics.hasNonNullValue()) {
+      if (statistics.hasNonNullValue() &&
+          createdBy != null &&
+          !shouldIgnoreStatistics(createdBy)) {
         stats.setMax(statistics.getMaxBytes());
         stats.setMin(statistics.getMinBytes());
-     }
+      }
     }
     return stats;
   }
 
+  private static boolean shouldIgnoreStatistics(String createdBy) {
+    final String[] versionTokens = createdBy.split(" ");
+
+    if (versionTokens.length < 3) {
+      return true;
+    }
+
+    final String app = versionTokens[0];
+    final String version = versionTokens[2].substring(0, 5).replaceAll("\\.", "");
+
+    if (app.equalsIgnoreCase("parquet-mr")) {
+      return Integer.parseInt(version) < STATISTICS_FIXED_VERSION;
+    }
+
+    return true;
+  }
+
```

Comment:
I would implement it slightly differently. In pseudo code:
```
if (!createdBy.startswith("parquet-mr")) {
  return false; // assume that other implementations got it right
}
return (version can be parsed) && (version >= STATISTICS_FIXED_VERSION)
```
Also make sure that you catch exceptions from `Integer.parseInt`